File: array/202.cpp
#include <bits/stdc++.h>

using namespace std;

int minProcessingTime(vector<int>& processorTime, vector<int>& tasks) {
    sort(processorTime.begin(), processorTime.end());
    sort(tasks.begin(), tasks.end(), greater<int>());
    int index = 0;
    int minTime = INT_MIN;

    for (int i = 0; i < processorTime.size(); ++i) {
        minTime = max(minTime, processorTime[i] + tasks[index]);
        index += 4;
    }

    return minTime;
}

int main() {
    vector<int> t = {2, 3, 1, 2, 5, 8, 4, 3};
    vector<int> pt = {10, 20};

    cout << minProcessingTime(pt, t);

    return 0;
}

File: array/202.py
from typing import List


def minProcessingTime(processorTime: List[int], tasks: List[int]) -> int:
    min_time = float("-inf")
    index = 0
    processorTime.sort()
    tasks.sort(reverse=True)

    for elem in processorTime:
        min_time = max(elem + tasks[index], min_time)
        index += 4

    return min_time


answer = minProcessingTime([8, 10], [2, 2, 3, 1, 8, 7, 4, 5])
print(answer)
